api/peer/doc.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// Package peer contains interfaces pertaining to peers, peer lists, peer list
+// updaters, and generally how to choose a peer for an outbound request.
+//
+// The `go.uber.org/yarpc/peer` package tree provides the corresponding
+// implementations.
+//
+// Some transports support the notion of peers.
+// A peer instance models the host and port of a remote listening socket for a
+// particular transport protocol, like HTTP and TChannel.
+// For example, YARPC might have a TChannel peer instance to track connections
+// to 127.0.0.1:4040.
+//
+// Some transports, like HTTP and TChannel support peer selection on their outbounds.
+// A `peer.Chooser` allows an outbound to obtain a peer for a given context and
+// request, and also manages the lifecycle of all components it needs to do so.
+// A peer chooser is typically also a `peer.List`, thus a `peer.ChooserList`.
+//
+// Peer list updaters send `Update` message to a `peer.List` to add and remove
+// tracked peer identifiers.
```

Comment:
Maybe change to "tracked peers."

These docs can talk in terms of high level entities rather than APIs.